llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp

```diff
@@ -401,6 +401,15 @@ RISCVRegisterBankInfo::getInstrMapping(const MachineInstr &MI) const {
   case TargetOpcode::G_SELECT: {
     LLT Ty = MRI.getType(MI.getOperand(0).getReg());
 
+    if (Ty.isVector()) {
```

Comment:
You could try
```
 GSelect *Sel = cast<GSelect>(&MI);
```
, but ...

Reply:
Are you suggesting: `MI.getOperand(2).getReg()` -> `GSelect *Sel = cast<GSelect>(&MI); MRI.getType(Sel->getCondReg())`?

Reply:
Exactly. Then you don't have to remember the operand indices and the code becomes more readable.